h/browser/chrome/lib/hypothesis-chrome-extension.js

```diff
@@ -147,10 +148,20 @@ function HypothesisChromeExtension(dependencies) {
       browserAction.deactivate(tabId);
     }
 
-    settings.then(function(settings) {
-      browserAction.updateBadge(tabId, tab.url, settings.serviceUrl);
+    // Here we're calling uriinfo() for two reasons:
+    // 1. Because we want to call updateBadge() in the then() function.
+    // 2. Because we want to fetch the info for the new URL asap, when we
+    //    call uriinfo() again later (when the user clicks on the browser
+    //    button to activate the sidebar) it will return the already-resolved
+    //    Promise.
+    uriinfo(tab.url).then(function(info) {
+      if (!info.blocked) {
+        browserAction.updateBadge(info.total, tab.id);
+      }
+    }).
+    catch(function() {
+      // Silence console error message about uncaught exception here.
```

Comment:
@robertknight How's this for the error handling? https://github.com/hypothesis/h/commit/6dabcb63be7213b2395a03ff51f19fa51c25a220

I'm a bit sad that I had to use `this`, but I think the alternative is just `return {name: 'UriInfoError', ...}` and then in the catch check the value of `.name`.